kiwi/builder/disk.py

```diff
@@ -793,14 +790,17 @@ def _create_dracut_config(self):
 
     def _create_system_dracut_config(self):
         dracut_modules = []
-        dracut_modules_omit = ['kiwi-live', 'kiwi-dump', 'kiwi-repart']
+        dracut_modules_omit = []
         if self.root_filesystem_is_overlay:
             dracut_modules.append('kiwi-overlay')
+            self._write_dracut_config(
+                config=[], modules=dracut_modules,
+                omit_modules=dracut_modules_omit
+            )
         else:
-            dracut_modules_omit.append('kiwi-overlay')
-        self._write_dracut_config(
-            config=[], modules=dracut_modules, omit_modules=dracut_modules_omit
-        )
+            Path.wipe(''.join(
+                [self.root_dir, Defaults.get_dracut_conf_name()]
+            ))
```

Comment:
I think deleting the dracut.conf is wrong. Actually the file is only written if the overlay module is in use. In case of the overlay module it has to stay active. If it's not taken into account again on e.g kernel-update the system will be broken. Our motivation should be to write the dracut config when needed and then it should also be persistently needed for the live time of the image

Reply:
I am not sure I follow this comment. With the code written here the dracut configuration is always deleted unless there is a root overlay, for which we need the kiwi-overlay module. I can improve the code above to make it more explicit. But the result in terms of the root-tree remains the same. After the build we need a persistent dracut configuration for the case we have a root overlay and for the case of live image (`builder/live.py` does not remove the dracut configuration). 

In that latter case I doubt it is strictly needed as I doubt the initrd can be updated, but I definitely prefer to keep it around since this would be the correct configuration in case initrd could be updated.

Reply:
done